### DISCUSSION TITLE
QSelect with dynamic items

### DISCUSSION THREAD

**Discussion by Allihopp** [quasarframework/quasar#10713] [General - Components / Directives / etc]
Hello everybody, I have a little problem.

First of all, this is my code for the client side:
```
<template>
  <q-page>
    <q-select
      v-for="e in equipment"
      :key="e.id"
      :label="e.label"
      :options="status"
      v-model="e.model"
    />
  </q-page>
</template>

<script>
import { defineComponent, ref, onMounted } from 'vue'
import { api } from 'boot/axios'

export default defineComponent({
  name: 'PageIndex',
  setup() {
    const status = ['Working', 'Not working']
    let equipmentList = []
    const equipment = ref(equipmentList)

    onMounted(() => {
      api
        .get('/equipment')
        .then((response) => {
          equipmentList = response.data.options
        })
        .catch((error) => {
          console.log(error)
        })
    })

    return {
      status,
      equipment
    }
  }
})
</script>
```
The server API responses with a json:
```
{
  "options": [
    {
      "id": 1,
      "label": "Item 1",
      "status": null
    },
    {
      "id": 2,
      "label": "Item 2",
      "status": null
    },
    {
      "id": 3,
      "label": "Item 3",
      "status": null
    }
  ]
}
```
**What it should do?** After user opens the page, client-side should recieve from server an array of data and render every one of it's item in separate QSelect. For the "options" client has static variable "status". When the user chooses an option in every QSelect, "equipment" array should update the "status" of every item.
_For example_: user loads the page, picks a "Not working" choice for "Item 2", and the "status" variable for this "Item 2" sets to "false".

**The problem** right now - nothing renders, I mean <q-select>, if it's working through the API request (console log shows everything it need). If the "options" set statically in the client code - everything works fine.

Can anyone help with this?

**Comment by ontwikkelfabriek** [comment]
@Allihopp 

Please instead of posting code listing (without proper JS code highlighting), create a codepen.io for better support. You can easily create a codepen.io by extending an example or this default pen:
https://codepen.io/rstoenescu/pen/xxEvBLR

**Reply by pikil** [reply]
+1

**Comment by pikil** [comment]
ref() creates a watcher for the data provided initially in it meaning that there might be no reactivity connected to equipmentList variable. Instead, try to create ```const equipment = ref([])``` and apply the response.data.options directly to the value ```equipment.value = response.data.options```

**Comment by metalsadman** [comment]
consider other peeps suggestions, then see https://codepen.io/metalsadman/pen/vYZdQzB?editors=1010
